Core/clim-core/frames.lisp

```diff
@@ -773,45 +768,44 @@ documentation produced by presentations.")
                         '((%pointer-documentation%
                            pointer-documentation-pane)))))
   `(defmethod generate-panes ((fm frame-manager) (frame ,class-name))
-     (let ((*application-frame* frame))
-       (with-look-and-feel-realization (fm frame)
-         (unless (frame-panes-for-layout frame)
-           (setf (frame-panes-for-layout frame)
-                 (list
-                  ,@(loop for (name . form) in panes
-                          collect `(cons ',name ,(generate-pane-creation-form
-                                                  name form))))))
-         (let ,(loop for (name . form) in panes
-                     collect `(,name (alexandria:assoc-value
-                                      (frame-panes-for-layout frame)
-                                      ',name :test #'eq)))
-           ;; [BTS] added this, but is not sure that this is correct for
-           ;; adding a menu-bar transparently, should also only be done
-           ;; where the exterior window system does not support menus
-           (setf (frame-panes frame)
-                 (ecase (frame-current-layout frame)
-                   ,@(if (or menu-bar pointer-documentation)
-                         (mapcar (lambda (layout)
-                                   `(,(first layout)
-                                     (vertically ()
-                                       ,@(cond
-                                           ((eq menu-bar t)
-                                            `((setf (frame-menu-bar-pane frame)
-                                                    (make-menu-bar ',class-name))))
-                                           ((consp menu-bar)
-                                            `((make-menu-bar
-                                               (make-command-table
-                                                nil :menu ',menu-bar))))
-                                           (menu-bar
-                                            `((make-menu-bar ',menu-bar)))
-                                           (t nil))
-                                       ,@(rest layout)
-                                       ,@(when pointer-documentation
-                                           '(%pointer-documentation%)))))
-                                 layouts)
-                         layouts)))))
-       ;; Update frame-current-panes and the special pane slots.
-       (update-frame-pane-lists frame))))
+     (with-look-and-feel-realization (fm frame)
+       (unless (frame-panes-for-layout frame)
```

Comment:
I thought about this change as well but should `with-look-and-feel-realization` actually bind `*application-frame*`?

Reply:
It is not said in the spec, but I believe that this is the right thing to do. Otherwise frame argument would be unnecessary. Moreover from more tangible perspective, it is important that *application-frame* is bound to the current frame if we want to be able to use i.e the instance slots in pane definition. Consider the following:

```common-lisp
(define-application-frame poem-viewer ()
  ((title  :initarg :title  :reader poem-title)
   (verses :initarg :verses :reader poem-verses))
  (:panes (app :application
               :display-function (lambda (frame pane)
                                   (format pane (poem-verses frame)))))
  (:layouts
   (default
    (labelling (:label (poem-title *application-frame*)) ; <----
      app))))

(run-frame-top-level
 (make-application-frame 'poem-viewer
                         :title "CLIM"
                         :verses "Roses are red,~
                                  Violets are blue,~
                                  CLIM is software,~
                                  Which does kaboom."))
```

Reply:
It is clear that `define-application-frame` has to generate code that binds `*application-frame*`, my question was whether `with-look-and-feel-realization` should do it.

The argument regarding the parameter makes sense to me.